Documentation/en-us/TestUsingTestDoubles.md

```diff
@@ -0,0 +1,131 @@
+# Testing with Mock
+
+## Test doubles
+
+Following problem comes up often when writing tests. Here, `B` Class uses(depends on) `A Class`.
+
+![](https://github.com/Quick/Assets/blob/master/Screenshots/TestUsingMock_BusesA.png)
+
+Here's one example: BTests tests B, which uses A. Now bugs in A cause BTests to fail, even though B is fine! 
+All the failures are confusing--it's hard to answer the question "what's broken?"
+
+To avoid this problem, we can use a stand-in for A in Btests. In this case, we call stand-in for A `A' class`.
+
+![](https://github.com/Quick/Assets/blob/master/Screenshots/TestUsingMock_BusesAmock.png)
+
+`A' class` has same methods--however implementation might be different, and properties as `A Class`. So we can replace `A Class` with `A' Class` in BTests.
+
+Objects like `A' Class` are called 'test doubles'. 'test doubles' are used as stand-in for tests.
+There are some kinds of 'test doubles'.
+
+- Mock object: used for receiving output from test class
+- Stub object: used for providing input to test class
+- Fake object: behaves similar as original class
+
+Here, we explain using `Mock` next section.
+
+## Mock
+
+Mock is object(class, struct) used for verifying test object works with other objects from point of view of test target's output.
+
+### Writing test with Mock in Swift
+
+#### Sample app
+
+For example, we show an app which retrieves data from internet.
```

Comment:
I would put this example, or any example, first in this document. That's because, as a reader, I'm very impatient. I don't think I would pay attention to explanations of what "test doubles" or "mocks" were unless I knew that I wanted to use them. So I think it'd be best to motivate the reader first, before explaining any technical jargon like "test double" or "mock".

Put another way: I think this document should answer the question "why should you, the reader, care about what test doubles are?" in as few sentences as possible at the very beginning. Then it should proceed to explain what test doubles are and how to use them.